samza-core/src/main/java/org/apache/samza/zk/ZkMetadataStore.java

```diff
@@ -103,6 +104,25 @@ public void init() {
     return null;
   }
 
+  private byte[] readFromZk(String zkPath) {
+    long retryTimeOutMs = zkConfig.getZkSessionTimeoutMs();
+    long startTimeMs = System.currentTimeMillis();
+    while (true) {
+      try {
+        byte[] value = zkClient.readData(zkPath);
+        LOG.info("Read for zkPath: {} returned: {}.", zkPath, value);
```

Comment:
printing value will just print the reference right?
if this is encoded, could you print it as a string instead.